core/templates/components/summary-tile/story-summary-tile.component.spec.ts

```diff
@@ -0,0 +1,448 @@
+    let circumference = (2 * 20 * Math.PI).toString();
+
+    component.ngOnInit();
+
+    expect(component.strokeDashArrayValues).toBe('');
+    expect(component.completedStrokeDashArrayValues).toBe('0 ' + circumference);
+
+    // StorySummary with 1 node with 1 completed stories.
+    component.storySummary = StorySummary.createFromBackendDict({
+      id: 'storyId',
+      title: 'Story Title',
+      node_titles: ['node1'],
+      thumbnail_filename: 'thumbnail.jpg',
+      thumbnail_bg_color: '#FF9933',
+      description: 'This is the story description',
+      story_is_published: true,
+      completed_node_titles: ['node1'],
+      url_fragment: 'story1',
+      all_node_dicts: []
+    });
+
+    component.ngOnInit();
+
+    expect(component.strokeDashArrayValues).toBe('');
+    expect(component.completedStrokeDashArrayValues).toBe('');
+
+    // StorySummary with 5 node with 3 completed stories.
+    component.storySummary = StorySummary.createFromBackendDict({
+      id: 'storyId',
+      title: 'Story Title',
+      node_titles: ['node1', 'node2', 'node3', 'node4', 'node5'],
+      thumbnail_filename: 'thumbnail.jpg',
+      thumbnail_bg_color: '#FF9933',
+      description: 'This is the story description',
+      story_is_published: true,
+      completed_node_titles: ['node1', 'node2', 'node3'],
+      url_fragment: 'story1',
+      all_node_dicts: []
+    });
+
+    component.ngOnInit();
+
+    // The values are calculated by the same formula as explained above.
+    expect(component.strokeDashArrayValues).toBe('20.132741228718345 5');
+    expect(component.completedStrokeDashArrayValues).toBe(
+      '20.132741228718345 5 20.132741228718345 5' +
+      ' 20.132741228718345 55.26548245743669');
+  });
+
+  it('should link chapter title to URL', () => {
+    component.storySummary = StorySummary.createFromBackendDict({
+      id: 'storyId',
+      title: 'Story Title',
+      node_titles: ['node1'],
+      thumbnail_filename: 'thumbnail.jpg',
+      thumbnail_bg_color: '#FF9933',
+      description: 'This is the story description',
+      story_is_published: true,
+      completed_node_titles: ['node1'],
+      url_fragment: 'story1',
+      all_node_dicts: [
+        {
+          id: 'node1',
+          title: 'Node 1',
+          description: 'This is node 1',
+          destination_node_ids: [],
+          prerequisite_skill_ids: [],
+          acquired_skill_ids: [],
+          outline: '',
+          outline_is_finalized: true,
+          exploration_id: null,
+          thumbnail_bg_color: null,
+          thumbnail_filename: null
+        }
+      ]
+    });
+
+    component.ngOnInit();
+
+    expect(component.getChapterUrl('Node which is not present')).toBe('');
+    expect(component.getChapterUrl('Node 1')).toBe(
+      '/explore/null?story_url_fragment=story1&topic_url_fragment=' +
+      'undefined&classroom_url_fragment=undefined&node_id=node1');
+  });
+
+  it('should link story to title and thumbnail', () => {
```

Comment:
Perhaps `should populate the story URL when URL fragments are set`?

Reply:
Done